### PR TITLE
Extend tests from release testing

### PR DESCRIPTION
**Pull request by serprex**
Will likely extend this PR with more test cases from release testing, so merging should wait until after release

### CODE REVIEW

**Comment by metdos** [comment]
> Will likely extend this PR with more test cases from release testing, so merging should wait until after release

I think as long as the PR get reviewed we can always open multiple PRs and merge them separately.

File: src/test/regress/expected/multi_multiuser.out [hunk 1]
```diff
@@ -536,7 +536,7 @@ ERROR:  could not receive file "base/pgsql_job_cache/job_0042/task_000001/p_0000
 -- different user should not be able to fetch partition file
 SET ROLE usage_access;
 SELECT worker_fetch_partition_file(42, 1, 1, 1, 'localhost', :worker_1_port);
-WARNING:  could not open file "base/pgsql_job_cache/job_0042/task_000001/p_00001.18007": No such file or directory
+WARNING:  could not open file "base/pgsql_job_cache/job_0042/task_000001/p_00001.17981": No such file or directory
```
Comment (marcocitus) [inline src/test/regress/expected/multi_multiuser.out]: Why did the user OID change here?

File: src/test/regress/expected/worker_check_invalid_arguments.out [hunk 2]
```diff
@@ -67,8 +67,27 @@ ERROR:  column name array size: 2 and type array size: 3 do not match
 SELECT worker_merge_files_into_table(:JobId, :TaskId,
        				     ARRAY['textcolumn', 'binarycolumn'],
 				     ARRAY['text', 'integer']);
-ERROR:  invalid input syntax for integer: "\x0b50"
+ERROR:  invalid input syntax for type integer: "\x0b50"
```
Comment (marcocitus) [inline src/test/regress/expected/worker_check_invalid_arguments.out]: is this a PG12 change?

Comment (serprex) [reply src/test/regress/expected/worker_check_invalid_arguments.out]: Yes, it gets normalized

File: src/test/regress/expected/multi_multiuser.out (hunk 1 above)
Comment (serprex) [reply src/test/regress/expected/multi_multiuser.out]: Probably from being run on pg12. [It gets normalized](https://github.com/citusdata/citus/blob/169d2f193fb8e58f5425ac51668e93404501fc52/src/test/regress/bin/normalize.sed#L72)

**Pull request merged**